metricbeat/module/system/raid/ioctl.go

```diff
@@ -0,0 +1,112 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//+build darwin freebsd linux openbsd
+
+package raid
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+//this is a function var that we can override for the sake of testing
+var makeNewDevice = func(dev string) (MDData, error) {
```

Comment:
I'd define a function, and assign it to the variable, this way we could still explicitly call to one or other implementation.

```suggestion
var NewMDDevice func(dev string) (MDData, error) = NewMDDevice

func NewMDDevice(dev string) (MDData, error) {
```